sympy/matrices/reductions.py

```diff
@@ -300,6 +310,10 @@ def _rref(M, iszerofunc=_iszero, simplify=False, pivots=True,
     of the matrix, set ``noramlize_last=False``
     """
 
+    types = list(map(type, M))
+    if all(issubclass(t, Expr) for t in types):
+        return _rref_DOM(M, pivots)
```

Comment:
Use `set` rather than `list`.

Reply:
Maybe there should be a helper function for checking that all elements are Expr. That could have a docstring that explains why the check is needed.